central/declarativeconfig/manager_impl.go

```diff
@@ -87,5 +109,64 @@ func (m *managerImpl) ReconcileDeclarativeConfigs(contents [][]byte) {
 			transformedConfigurations[protoType] = append(transformedConfigurations[protoType], protoMessages...)
 		}
 	}
-	// No-op, see the TODOs within the function comment.
+
+	m.transformedMessagesMutex.Lock()
+	defer m.transformedMessagesMutex.Unlock()
+	m.transformedMessagesByHandler[handlerID] = transformedConfigurations
+	m.shortCircuitReconciliationLoop()
+}
+
+// shortCircuitReconciliationLoop will short circuit the reconciliation loop and trigger a reconciliation loop run.
+// Note that the reconciliation loop will not be run if:
+//   - the short circuit loop signal has not been reset yet and is de-duped.
+//   - a current reconciliation loop run is in progress.
+func (m *managerImpl) shortCircuitReconciliationLoop() {
+	// In case the signal is already triggered, the current call (and the Signal() call) will be effectively de-duped.
+	m.shortCircuitSignal.Signal()
+}
+
+func (m *managerImpl) startReconciliationLoop() {
+	m.reconciliationTicker = time.NewTicker(m.reconciliationTickerDuration)
+
+	go m.reconciliationLoop()
+}
+
+func (m *managerImpl) reconciliationLoop() {
+	// While we currently do not have an exist in the form of "stopping" the reconciliation, still, ensure that
+	// the ticker is stopped when we stop running the reconciliation.
+	defer m.reconciliationTicker.Stop()
+	for {
+		select {
+		case <-m.shortCircuitSignal.Done():
+			m.shortCircuitSignal.Reset()
+			m.runReconciliation()
+		case <-m.reconciliationTicker.C:
+			m.runReconciliation()
+		}
+	}
+}
+
+func (m *managerImpl) runReconciliation() {
+	// We shouldn't trigger a parallel reconciliation run, hence we should ensure that the flag is not set to true.
```

Comment:
I just now realised that we can't trigger parallel runs either way - all `runReconciliation` calls are happening in the one goroutine. Perhaps I'm missing something?